compiler/shared/main/scala/mlscript/compiler/codegen/CppAst.scala

```diff
@@ -0,0 +1,198 @@
+package mlscript.compiler.codegen.cpp
+
+import mlscript._
+import mlscript.utils._
+import mlscript.utils.shorthands._
+import mlscript.compiler.utils._
+
+enum Specifier:
+  case Extern
+  case Static
+  case Inline
+
+  def toDocument = this match
+    case Extern => "extern" |> raw 
+    case Static => "static" |> raw
+    case Inline => "inline" |> raw
+
+  override def toString(): Str = toDocument.print
```

Comment:
In Scala, it's idiomatic that methods assumed to be without side effects do not take an extra pair of empty parens.
```suggestion
  override def toString: Str = toDocument.print
```
The same remark applies everywhere else you do this.